### DISCUSSION TITLE
分担

### DISCUSSION THREAD

**Discussion by torus** [torus/42-minishell#2] [General]
![Screenshot from 2021-03-18 17-22-10](https://user-images.githubusercontent.com/65044/111596273-531a0300-8810-11eb-83c3-311dc4843947.png)
↑この絵でいうと、どの部分が作りたいですか？

**Comment by JUNNETWORKS** [comment]
個々の要素の詳細がわからないので, まだなんとも言えないです...
ちなみにこの図が載ってる本ってなんですか?

**Reply by torus** [reply]
https://m-takagi.github.io/aosa-ja/aosa.pdf
これです。Bash 以外にもいろいろ載ってます。
nfukada さんに教えてもらいました。

**Comment by JUNNETWORKS** [comment]
メモ: 2021/03/27 MTG

- Lexical analysis and ... : パーサー
- Expansion: 変数などを展開するところ
  - BraceExpansion: 中括弧が使われた時に展開する
  - TildeExpansion: ホームディレクトリを表す(`~`)を展開する
  - Variable and parameter... : 変数を展開する
  - Word Spliting: 展開した文字列にスペースが含まれていた場合の処理
  - FileNameGeneration: ワイルドカード(`*`)によるファイル名展開
- CommandExcecution: 実際に実行するところ
  - 子プロセスの生成実行
  - ビルドインコマンドの実行

ExpansionはMandatoryパートだと "Variable and parameter..." と "Word Splitting" だけやれば良さそう

大雑把な分担
- パーサー&Expansion: ひさいさん
- Command Execution: JUN

**Reply by JUNNETWORKS** [reply]
開発形式はお互いにPRを送り合って, もう一人の人がそのコードを見て問題無さそうならマージするという感じ.

できればユニットテストを書きたい.

タスクがある程度具体的になってきたらGitHubのProjects機能で管理する.

**Reply by JUNNETWORKS** [reply]
スケジュール

ひさいさん
- 木曜日以外なら大体いつでも大丈夫です.
- 昼間の時間が好ましい.
- 夜間は出来ない.

JUN
- バイトやめるまでの1~2週間くらいの間は月水金忙しい
- それ以外の日は基本暇
- バイト辞めたら毎日42やるからいつでもOK

**Reply by JUNNETWORKS** [reply]
[決定] 毎週土曜日15時くらいにボイチャつないで作業したり雑談したりする.

**Reply by JUNNETWORKS** [reply]
`main()` と ヘッダーファイル と Makefile は今日作って作業に取り掛かりやすい状態にしたい.

**Reply by JUNNETWORKS** [reply]
Norm対応をどうするか
- [normV3](https://github.com/42School/norminette)を使う.
- 42ヘッダーをファイル作成時点で付ける.
- PRを出す時点で以下の状態にする
  - Normの "1ファイル内の関数の数" 以外の項目をクリアした状態にする.

**Reply by JUNNETWORKS** [reply]
仕様やボイチャの議事録などは hackmd を使って記録する.

**Reply by JUNNETWORKS** [reply]
コミットについて

- 日本語で書く
- 1コミットが1つの変更になるようにする. 関係ない変更を同じコミットに含めない

**Reply by JUNNETWORKS** [reply]
ユニットテストはファイル(機能)ごとに作る.
例: `parser.c` のテストは `parser_test.c` に書く

ユニットテストを見ればその関数の仕様がわかるようにテストを書く.

ユニットテストを一括実行出来るMakefileのルールを作っておく.

**Reply by JUNNETWORKS** [reply]
ブランチは機能(feature)ごとに作る. 例: `feature/parser`

種類は
- `feature`
- `fix`
- `refactor`

`develop`ブランチは作らず, PRは`main`ブランチに投げる.

**Reply by JUNNETWORKS** [reply]
関数名の命名規則

関数名の先頭に簡単なプレフィックスを付けておく

例; CommandExecution関連なら `exec_hogehoge()`

**Reply by torus** [reply]
> ユニットテストはファイル(機能)ごとに作る.
> 例: `parser.c` のテストは `parser_test.c` に書く

ユニットテストのコードはたぶんマクロとか使いまくりで Norm に適合しないので、`test` というディレクトリに入れることにしましょうか。

**Reply by JUNNETWORKS** [reply]
もしくは以下のようにするとテストコード以外の.cファイルと.hファイルをNorminetteにかけることが出来ます.

```bash
norminette $(find . -type f -regextype posix-basic ! -regex ".*_test\.\(c\|h\)" -regex ".*\.\(c\|h\)")
```
